torch/_jit_internal.py

```diff
@@ -650,26 +651,25 @@ def safe_is_subclass(the_type, super_type):
 
     return optional or union_optional
 
-# fake Python container type for Future/RRef
-T = TypeVar('T')
-
-class Future(Generic[T]):
-    __slots__ = ['__args__']
-
-    def __init__(self, types):
-        self.__args__ = types
-
 def is_future(ann):
     if ann is Future:
-        raise RuntimeError('Attempted to use torch.jit.Future without a '
-                           'contained type. Please add a contained type, e.g. '
-                           'torch.jit.Future[int]')
+        raise RuntimeError(
+            "Attempted to use torch.jit.Future without a "
+            "contained type. Please add a contained type, e.g. "
+            "torch.jit.Future[int]"
+        )
     return getattr(ann, "__origin__", None) is Future
 
 if torch.distributed.rpc.is_available():
     from torch.distributed.rpc import RRef
 
     def is_rref(ann):
+        if ann is RRef:
+            raise RuntimeError(
+                "Attempted to use torch.jit.RRef without a "
+                "contained type. Please add a contained type, e.g. "
+                "torch.jit.RRef[int]"
```

Comment:
ditto

Reply:
For this it's fine to use `torch.distributed.rpc.RRef` as no BC issue involves.

Reply:
Let me change it to a bare `RRRef` to avoid showing bias to users.